src/Renci.SshNet/ForwardedPortLocal.cs

```diff
@@ -103,7 +103,7 @@ public ForwardedPortLocal(string boundHost, uint boundPort, string host, uint po
             Host = host;
             Port = port;
             _status = ForwardedPortStatus.Stopped;
-            _logger = SshNetLoggingConfiguration.LoggerFactory.CreateLogger<ForwardedPortLocal>();
+            _logger = Session.SessionLoggerFactory.CreateLogger<ForwardedPortLocal>();
```

Comment:
I have kicked off a test run. I think it is going to show NullReferenceException from this line (lots of files are still not null-annotated)

Reply:
This is hard to solve without a ?. For now i changed the code to create the logger only when logging is needed. Because the `Session` property could change during the lifetime of the `ForwardedPort` I chose not to cache the logger, but get an instance only when needed. Looking at the code this should only occur in exceptional cases.